app.rb

```diff
@@ -0,0 +1,186 @@
+  age = nil
+  name = nil
+  parent_permission = nil
+  classroom_type = nil
+
+  loop do
+    puts 'Age:'
+    age = gets.chomp
+
+    puts 'Name:'
+    name = gets.chomp
+
+    puts 'Has parent permission? [Y/N]:'
+    parent_permission = gets.chomp
+
+    puts 'Classroom type eg. Math class:'
+    classroom_type = gets.chomp
+
+    break unless age.empty? || name.empty? || classroom_type.empty? || parent_permission.empty?
+  end
+
+  resume(age, name, parent_permission, classroom_type)
+end
+
+# create a teacher block .....
+def create_teacher
+  age = nil
+  name = nil
+  specialization = nil
+
+  loop do
+    puts 'Age:'
+    age = gets.chomp
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Specialization:'
+    specialization = gets.chomp
+
+    break unless age.empty? || name.empty? || specialization.empty?
+
+    puts 'Please enter a valid value for all fields'
+  end
+
+  person = Person.new(age, name)
+  new_teacher = Teacher.new(person, specialization)
+  Teacher.teachers << new_teacher
+  puts 'Teacher created successfully'
+end
+
+# list all people block...
+def retrieve_teachers
+  teachers = Teacher.teachers
+  puts '' if teachers.empty?
+  teachers
+end
+
+def retrieve_students
+  students = Classroom.students
+  puts '' if students.empty?
+  students
+end
+
+def all_people
+  people = []
+  all_teachers = retrieve_teachers
+  all_students = retrieve_students
+
+  all_teachers.each do |teacher|
+    teacher_str = "[Teacher] Name: #{teacher.name}, Age: #{teacher.age}, Id: #{teacher.id}"
+    people << { str: teacher_str, obj: teacher }
+  end
+
+  all_students.each do |student|
+    student_str = "[Student] Name: #{student.name}, Age: #{student.age}, Id: #{student.id}"
+    people << { str: student_str, obj: student }
+  end
+
+  if people.empty?
+    puts 'Empty: Please create a person.'
+  else
+    people.each do |human|
+      puts human[:str]
+    end
+  end
+  people
+end
+
+def add_rentee(book_list, persons, book_num)
+  rented_book = book_list[book_num]
+
+  puts 'Select a person from the following list by number (not id)'
+  persons.each_with_index do |person, index|
+    puts "#{index}) #{person[:str]}"
+  end
+
+  rentee = gets.chomp.to_i
+  receiver = persons[rentee][:obj]
+
+  puts 'Date:'
+  date_rented = gets.chomp
+  receiver.add_rental(rented_book, date_rented)
+
+  puts 'Rental created successfully'
+end
+
+# create rental block...
+def create_rental
+  book_list = Book.saved_books
+  persons = all_people
+  if book_list.empty?
+    puts 'No books! please add a book'
+  elsif persons.empty?
+    puts 'puts please go back and create a person to rent the book'
+  else
+    puts 'Select a book from the following list by number'
+
+    book_list.each_with_index do |book, index|
+      puts "#{index}) Title: #{book.title}, Author: #{book.author}"
+    end
+    book_num = gets.chomp.to_i
+
+  end
+  add_rentee(book_list, persons, book_num)
+end
+
+def user_rentals
+  puts 'ID of person:'
+  id = gets.chomp.to_i
```

Comment:
**[OPTIONAL]**

- [ ] Kudos 👏 on implementing the requirements and additional feature of classroom 😊 It would be nice if you show all the users id before taking id as input this is the user_rental method because how a user is supposed to know which id belongs to who so it would be better if you all people name with their respective ids 👍

Reply:
thank you i will do that.